sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -47,7 +47,8 @@ private[hive] case class HiveSimpleUDF(
   with HiveInspectors
   with CodegenFallback
   with Logging
-  with UserDefinedExpression {
+  with UserDefinedExpression
+  with ImplicitCastInputTypes {
```

Comment:
`ScalaUDF` doesn't extend `ImplicitCastInputTypes` either. Does it have the same problem?

Reply:
No, `ImplicitTypeCasts` has checked `ScalaUDF ` like `case udf: ScalaUDF if udf.inputTypes.nonEmpty =>`